### PR TITLE
Write classification model output to a file for later use in reports.

### PR DESCRIPTION
**Pull request by Shruthi42**
A file `model_outputs.csv` is written with columns `subject`, `prediction_target`, `label`, `model_output` and `cross_validation_split`. This file is not written out for sequence models.

This PR also changes the signature for the function `generate_custom_report` in `ModelConfigBase`.

### CODE REVIEW

**Review by melanibe — APPROVED** [review]

File: InnerEye/ML/model_testing.py [hunk 1]
```diff
@@ -409,72 +410,79 @@ def classification_model_test(config: ScalarModelBase,
...
+    checkpoint_paths = checkpoint_handler.get_checkpoints_to_test()
+    if not checkpoint_paths:
         raise ValueError("There were no checkpoints available for model testing.")
 
-    result = test_epoch(checkpoint_paths=checkpoints_to_test)
-    if result is None:
+    pipeline = create_inference_pipeline(config=config,
+                                         checkpoint_paths=checkpoint_paths)
+    if pipeline is None:
         raise ValueError("There was no single checkpoint file available for model testing.")
```
Comment (ozan-oktay) [inline InnerEye/ML/model_testing.py:420]: Maybe update the exception message "inference pipeline couldn't be created" or sth like that

Comment (Shruthi42) [reply InnerEye/ML/model_testing.py]: Done!

File: InnerEye/ML/model_testing.py [hunk 2]
```diff
@@ -409,72 +410,79 @@ def classification_model_test(config: ScalarModelBase,
...
+
+    # for mypy
+    assert isinstance(pipeline, ScalarInferencePipelineBase)
+
+    ml_util.set_random_seed(config.get_effective_random_seed(), "Model Testing")
+    ds = config.get_torch_dataset_for_inference(data_split).as_data_loader(
+        shuffle=False,
+        batch_size=1,
```
Comment (ozan-oktay) [inline InnerEye/ML/model_testing.py:428]: I think the batch size was set to 1 in the past and it is not related to this PR. However, if you think that inference metric computation takes long, we could do the inference, metric computation, and logging in batches as well.

**Review by ozan-oktay — APPROVED** [review]
Thanks!

**Review by melanibe — APPROVED** [review]

**Pull request merged**